src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -4737,7 +4796,9 @@ void MergeTreeData::checkAlterPartitionIsPossible(
                         throw DB::Exception(ErrorCodes::SUPPORT_IS_DISABLED, "Only support DETACH PARTITION ALL currently");
                 }
                 else
-                    getPartitionIDFromQuery(command.partition, getContext());
```

Comment:
Need to be careful with possible side effects because other code paths rely on this method